cmd/soci/commands/ztoc/list.go

```diff
@@ -30,25 +34,69 @@ var listCommand = cli.Command{
 	Description: "list ztocs",
 	Flags: []cli.Flag{
 		cli.StringFlag{
-			Name:  "digest",
+			Name:  "ztoc-digest",
 			Usage: "filter ztocs by digest",
 		},
+		cli.StringFlag{
+			Name:  "image-ref",
+			Usage: "filter ztocs to those that are associated with a specific image",
+		},
 	},
 	Action: func(cliContext *cli.Context) error {
 		db, err := soci.NewDB()
 		if err != nil {
 			return err
 		}
-		digest := cliContext.String("digest")
+		ztocDgst := cliContext.String("ztoc-digest")
+		imgRef := cliContext.String("image-ref")
 
 		var artifacts []*soci.ArtifactEntry
-		db.Walk(func(ae *soci.ArtifactEntry) error {
-			if ae.Type == soci.ArtifactEntryTypeLayer &&
-				(digest == "" || ae.Digest == digest) {
-				artifacts = append(artifacts, ae)
+		if imgRef == "" {
```

Comment:
What is the expected behavior when both are present?  Does it make sense to split the two into private functions?

Reply:
Updated the PR to include the case where both --image-ref and --ztoc-digest are present. We will validate if the specified ztoc is with the specified image. If the ztoc is not with the image, the command will return an error.